### PR TITLE
Standardize API naming

### PR DESCRIPTION
**Pull request by cmatheson**
## Description

- rename `accessToken` to `access_token`
- Make SSO interface properties snake_case
- SSO.getAuthorizationURL -> SSO.getAuthorizationUrl
- Make MFA interfaces use snake_case
- Make admin portal interfaces use snake_case
- Make passwordless interfaces use snake_case
- Rename User to DirectoryUser

## Documentation

Does this require changes to the WorkOS Docs? E.g. the [API Reference](https://workos.com/docs/reference) or code snippets need updates.

```
[X] Yes
```

If yes, link a related docs PR and add a docs maintainer as a reviewer. Their approval is required.

### CODE REVIEW

File: src/directory-sync/interfaces/directory_user.interface.ts [hunk 1]
```diff
@@ -24,8 +24,8 @@ export interface User<
   state: 'active' | 'inactive' | 'suspended';
 }
 
-export interface UserWithGroups<
+export interface DirectoryUserWithGroups<
```
Comment (cmatheson) [inline src/directory-sync/interfaces/directory_user.interface.ts:27]: is there any reason to keep this separate `DirectoryUserWithGroups` interface around? seems like all uses of `DirectoryUser` are actually `DirectoryUserWithGroups`

**Comment by maxdeviant** [comment]
If we're standardizing on the casing, should we be standardizing on camelCase and then converting to and from snake_case at the API boundary?

It's a bit odd that we have any snake_case identifiers in the SDK's public API, as it's not a standard casing for Node packages.

File: src/mfa/interfaces/challenge-factor-options.ts [hunk 2]
```diff
@@ -1,8 +1,4 @@
-export type ChallengeFactorOptions =
-  | {
-      authenticationFactorId: string;
-    }
-  | {
-      authenticationFactorId: string;
-      smsTemplate: string;
-    };
+export interface ChallengeFactorOptions {
```
Comment (cmatheson) [inline src/mfa/interfaces/challenge-factor-options.ts:1]: pretty sure this is equivalent but feel free to disagree!

Comment (maxdeviant) [reply src/mfa/interfaces/challenge-factor-options.ts:1]: I think we should keep it as a union type, as these are mutually exclusive options that could diverge more longer-term.

**Comment by cmatheson** [comment]
> If we're standardizing on the casing, should we be standardizing on camelCase and then converting to and from snake_case at the API boundary?

Hmm, the upgrade path for existing people is nicer if we continue to use snake_case, but maybe that's not a good enough reason to keep preserve snake case.

**Comment by maxdeviant** [comment]
> > If we're standardizing on the casing, should we be standardizing on camelCase and then converting to and from snake_case at the API boundary?
> 
> Hmm, the upgrade path for existing people is nicer if we continue to use snake_case, but maybe that's not a good enough reason to keep preserve snake case.

A couple of thoughts here from the lens of us wanting to eventually end up using camelCase across the board:

- There are some places where we're currently using camelCase that are being rewritten to snake_case in this PR. This seems like a step backwards for these names
- Newer APIs (User Management) should be introduced using camelCase so that it doesn't involve a migration later
- Short of writing and shipping a codemod, there will have to be some point where consumers switch over to the camelCased APIs, so I don't see a strong reason why that switch shouldn't take place in v3

**Comment by oliverzheng** [comment]
I also prefer camelCasing in Node SDKs. What do we do with API responses that are snake_cased? Without doing anything about that, we'd have inconsistency within the usage of the SDK itself, where callsites would pass in camelCasedParams and get back snake_cased_properties all within the same code-block. 

As a point of reference, Stripe's Node SDK uses snake_casing for both params and responses: https://stripe.com/docs/api/credit_notes/create?lang=node

**Comment by maxdeviant** [comment]
> I also prefer camelCasing in Node SDKs. What do we do with API responses that are snake_cased? Without doing anything about that, we'd have inconsistency within the usage of the SDK itself, where callsites would pass in camelCasedParams and get back snake_cased_properties all within the same code-block.

We could add a translation layer where we translate the snake_cased responses to camelCase. This is what our other SDKs do as well (they just typically have built-in support for it in their JSON serialization library).

We'll already need to do this anyways for normalizing some of the enum values (mentioned in the Notion doc about SDK changes).

**Comment by cmatheson** [comment]
closing this branch (we should do camelCase instead of snake_case as done here).

**Pull request closed without merging**